ui/src/buckets/components/BucketList.tsx

```diff
@@ -131,38 +105,47 @@ class BucketList extends PureComponent<Props & WithRouterProps, State> {
     bucket: PrettyBucket,
     dataLoaderType: DataLoaderType
   ) => {
-    this.props.onSetBucketInfo(
+    const {onSetBucketInfo, onSetDataLoadersType, router} = this.props
+    onSetBucketInfo(
       bucket.organization,
       bucket.organizationID,
       bucket.name,
       bucket.id
     )
 
-    this.props.onSetDataLoadersType(dataLoaderType)
-
     this.setState({
       bucketID: bucket.id,
-      dataLoadersOverlayState: OverlayState.Open,
-    })
-  }
-
-  private handleDismissDataLoaders = () => {
-    this.setState({
-      bucketID: '',
-      dataLoadersOverlayState: OverlayState.Closed,
     })
-  }
 
-  private get isDataLoadersWizardVisible(): boolean {
-    const {bucketID, dataLoadersOverlayState} = this.state
-    return !!bucketID && dataLoadersOverlayState === OverlayState.Open
+    onSetDataLoadersType(dataLoaderType)
+    router.push(this.overlayRoute(dataLoaderType, bucket.id))
   }
 
   private get isBucketOverlayVisible(): boolean {
     const {bucketID, bucketOverlayState} = this.state
     return !!bucketID && bucketOverlayState === OverlayState.Open
   }
 
+  private overlayRoute(
```

Comment:
This seems a bit like a router inside a router

Reply:
yes, that makes sense. I'll change that